addons/account/models/account_move_line.py

```diff
@@ -2214,6 +2214,13 @@ def _check_amls_exigibility_for_reconciliation(self, shadowed_aml_values=None):
                                     This is usefull if you want to preview the reconciliation before doing some changes
                                     on amls like changing a date or an account.
         """
+        partial_matching_numbers = self.filtered(lambda aml: aml.matching_number and aml.matching_number.startswith("P")).mapped("matching_number")
+        if partial_matching_numbers:
+            for pmn in partial_matching_numbers:
+                if False not in self.filtered(lambda aml: aml.matching_number == pmn).mapped("reconciled"):
+                    raise UserError(_("You can only reconcile partially reconciled entries if one of them is not fully reconciled."))
+        self = self.filtered(lambda aml: not aml.reconciled or aml.matching_number not in partial_matching_numbers)
```

Comment:
```suggestion
        partial_matching_numbers = set(self
            .filtered(lambda aml: not aml.reconciled and aml.matching_number and aml.matching_number.startswith('P'))
            .mapped('matching_number'))
        )
        self = self.filtered(lambda aml: not aml.reconciled or aml.matching_number not in partial_matching_numbers)
```
We already gave you multiple cases to cover with your test cases. Add them.
CASE 1:
aml1 1000 1234P reconciled=True
aml2 -999 1234P reconciled=False
aml3 -1 reconciled=False
=> aml1 should be filtered automatically, no error

CASE 2:
aml1 1000 1234P reconcile=True
aml2 -999 reconciled=False
aml3 -1 reconciled=False
=> aml1 should not be filtered out, it must raise!

Reply:
If you meant modifying the tests for those two use cases, it is actually already the case. I have renamed the amls in the test to fit your examples so it's clearer. Your case 2 is tested on line 5571 with the assertRaises and your case 1 is tested right after that on line 5575.

If you meant something else than the tests, i didn't get it.